### DISCUSSION TITLE
Do I have to be the owner of GitHub repo to bridge from?

### DISCUSSION THREAD

**Discussion by mcepl** [git-bug/git-bug#902] [Q&A] [answered]
```
stitny~/b/n/v/vis (bugs)$ git bug bridge configure
[1]: github
[2]: gitlab
[3]: jira
[4]: launchpad-preview
target: 1
name [default]: martanne
Github project URL: https://github.com/martanne/vis
Github login: mcepl
Error: unknown credential type ""
stitny~/b/n/v/vis (bugs)$ 
```

I would like just to have read-only local repository of bugs from that project. Could I do it?

**Comment by MichaelMure** [comment, marked as answer]
As long as you have read access to that repo (for example if it's public), yes.

**Reply by mcepl** [reply]
So, what’s going on?

Also:

```
stitny~/b/n/v/vis (bugs)$ git bug bridge configure \
                              --name=martanne \
                              --target=github \
                              --token=ghp_somethingVerySecret \
                              --owner=martanne --project=vis
Successfully configured bridge: martanne
stitny~/b/n/v/vis (bugs)$ git bug bridge pull
Error: unknown credential type ""
stitny~/b/n/v/vis (bugs)$ 
```

**Reply by MichaelMure** [reply]
What version are you using? Could you try building from master?

**Reply by mcepl** [reply]
Well, if somebody, no names mentioned here, made releases more often then when he has 823 commits on master since the last one, his users wouldn’t use ancient versions. Let me try master …

**Reply by MichaelMure** [reply]
Haha, fair. I also wish I had more time to work on it ..

**Reply by mcepl** [reply]
Actually, I have already been using [git-bug](https://build.opensuse.org/package/show/devel:tools:scm/git-bug) from the commit 0eef939, which should be the latest master (I am the maintainer of the package).

**Reply by MichaelMure** [reply]
That looks like an actual bug then. Odd that it would surface now though, almost nothing changed on the github bridge for a while. It shouldn't be difficult to find out what's happening with a debugger.

**Reply by MichaelMure** [reply]
Hmmm, it might be that you have an old config for the bridge and it get confused. https://github.com/MichaelMure/git-bug-migration should fix that, but otherwise you can clean your git config manually.

Ideally git-bug would not choke on that, but that another story.

**Reply by mcepl** [reply]
I don’t think that’s the problem, I have never used git-bug before. Anyway, that would be the `git-bug` section of `.git/config`, right? Anywhere else?

Aren’t tokens somehow limited just to owner’s repositories? (Yes, I have `repo:public_repo` enabled, do I need something else?)

**Reply by mcepl** [reply]
And of course:

```
stitny~/b/n/v/vis (devel)$ /tmp/git-bug-migration_linux_amd64 --for-real
Migration #1
Purpose: Convert legacy identities into a complete data structure in git



Migration #2
Purpose: Migrate bridge credentials from the global git config to a keyring

Error applying migration: unknown credential type ""
stitny~/b/n/v/vis (devel)$ 
```

**Reply by MichaelMure** [reply]
It seems like you have a corrupted credential in the keyring, which is going to be located in a shared place (that is cross repo) on you machine. Where exactly it is depends on your OS:

> On Unix systems, it returns $XDG_CONFIG_HOME as specified by https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html if non-empty, else $HOME/.config. On Darwin, it returns $HOME/Library/Application Support. On Windows, it returns %AppData%. On Plan 9, it returns $home/lib.

**Reply by mcepl** [reply]
I see. Then that’s an explanation, but still not help. I use https://github.com/languitar/pass-git-helper, but `git credential fill` works as expected.

**Reply by MichaelMure** [reply]
That's a different thing. Git-bug stores credentials on its own as there is no easy cross platform solution to use that in git.
You can just delete that file and thing should go back to normal.

**Reply by mcepl** [reply]
The directory is `~/.config` on Linux, but I still don’t see any file having `git-bug*` in its name there. How should it be called?

**Reply by mcepl** [reply]
Of course, `~/.config/git/config` … I didn’t understand that you so mercilessly dump all that junk there. Sorry, when eliminating any mention of git-bug anywhere close to git, I have managed to do `git bug bridge pull martanne`. I will report if anything more noteworthy happens.

**Reply by mcepl** [reply]
> Hmmm, it might be that you have an old config for the bridge and it get confused. https://github.com/MichaelMure/git-bug-migration should fix that, but otherwise you can clean your git config manually.
> 
> Ideally git-bug would not choke on that, but that another story.

Talking about choking (`git-bug-migration` binary from the GitHub repository):

```
Migration #2
Purpose: Migrate bridge credentials from the global git config to a keyring

18ed7db: panic: crypto/hmac: hash generation function does not produce unique values

goroutine 1 [running]:
crypto/hmac.New(0xc00018d488, {0xc00039d140, 0x7, 0x8})
	/usr/lib/go-1.17/src/crypto/hmac/hmac.go:143 +0x292
github.com/dvsekhvalnov/jose2go/kdf.DerivePBKDF2({0xc00039d140, 0x3, 0x28}, {0xc00003f890, 0x1f, 0x30}, 0xa, 0x80, {0xd585d8, 0xc00046e300})
	/home/michael/go/pkg/mod/github.com/dvsekhvalnov/jose2go@v0.0.0-20180829124132-7f401d37b68a/kdf/pbkdf2.go:15 +0xb9
github.com/dvsekhvalnov/jose2go.(*Pbse2HmacAesKW).WrapNewKey(0xc00000e150, 0xc0000a52c0, {0xb3a480, 0xc00014c690}, 0xc0004c16a0)
	/home/michael/go/pkg/mod/github.com/dvsekhvalnov/jose2go@v0.0.0-20180829124132-7f401d37b68a/pbse2_hmac_aeskw.go:50 +0x339
github.com/dvsekhvalnov/jose2go.encrypt({0xc00017c340, 0x19b, 0x1a0}, 0x3, {0xb3a480, 0xc00014c690})
	/home/michael/go/pkg/mod/github.com/dvsekhvalnov/jose2go@v0.0.0-20180829124132-7f401d37b68a/jose.go:305 +0x1af
github.com/dvsekhvalnov/jose2go.EncryptBytes({0xc00017c340, 0x19b, 0x1a0}, {0xc5e501, 0x12}, {0xc586ea, 0x7}, {0xb3a480, 0xc00014c690}, {0xc00018d930, ...})
	/home/michael/go/pkg/mod/github.com/dvsekhvalnov/jose2go@v0.0.0-20180829124132-7f401d37b68a/jose.go:222 +0x3a5
github.com/dvsekhvalnov/jose2go.Encrypt(...)
	/home/michael/go/pkg/mod/github.com/dvsekhvalnov/jose2go@v0.0.0-20180829124132-7f401d37b68a/jose.go:187
github.com/99designs/keyring.(*fileKeyring).Set(0xc000621260, {{0xc00030c280, 0x45}, {0xc0000cc210, 0xa6, 0xb0}, {0x0, 0x0}, {0x0, 0x0}, ...})
	/home/michael/go/pkg/mod/github.com/99designs/keyring@v1.1.5/file.go:144 +0x1e9
github.com/MichaelMure/git-bug-migration/migration2/after/bridge/core/auth.Store({0x7f38bcba23a0, 0xc0004ad300}, {0xd62c18, 0xc00000edf8})
	/home/michael/dev/git-bug-migration/migration2/after/bridge/core/auth/credential.go:211 +0x4dd
github.com/MichaelMure/git-bug-migration/migration2.(*Migration2).migrate(0xc0003b81b0, {0xd6b640, 0xc00020e360}, {0xd6ad68, 0xc0004ad300})
	/home/michael/dev/git-bug-migration/migration2/migration2.go:85 +0x7a5
github.com/MichaelMure/git-bug-migration/migration2.(*Migration2).Run(0xd443c0, {0xc0003b81b0, 0x25})
	/home/michael/dev/git-bug-migration/migration2/migration2.go:31 +0x8e
main.runRootCmd(0xc0001a5e90, {0xc8}, {0xc0001a5ec0, 0x3, 0x0})
	/home/michael/dev/git-bug-migration/root.go:76 +0x383
main.NewRootCommand.func1(0xc0000d5080, {0xc58caf, 0x1, 0x1})
	/home/michael/dev/git-bug-migration/root.go:44 +0x31
github.com/spf13/cobra.(*Command).execute(0xc0000d5080, {0xc000032050, 0x1, 0x1})
	/home/michael/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000d5080)
	/home/michael/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x3ad
github.com/spf13/cobra.(*Command).Execute(...)
	/home/michael/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
	/home/michael/dev/git-bug-migration/main.go:6 +0x1e
stitny~/b/n/v/vis (devel)$ 
```

**Reply by MichaelMure** [reply]
Could you open separate issues for those? This discussion is not the place for that. Those will need to be fixed before a new release.

**Comment by mcepl** [comment]
This is probably long obsolete … interactive configuration of bridge (especially manipulation with tokens) is still not ideal, but with the non-interactive use, I have managed to make it work with 0.7.2.
